src/VisualStudio/Xaml/Impl/Implementation/LanguageServer/XamlLanguageServerProtocol.cs

```diff
@@ -23,25 +25,42 @@ namespace Microsoft.VisualStudio.LanguageServices.Xaml.LanguageServer
     internal sealed class XamlLanguageServerProtocol : AbstractRequestHandlerProvider
     {
         private readonly XamlProjectService _projectService;
+        private readonly IXamlLanguageServerFeedbackService? _feedbackService;
 
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public XamlLanguageServerProtocol([ImportMany] IEnumerable<Lazy<IRequestHandler, IRequestHandlerMetadata>> requestHandlers, XamlProjectService projectService)
+        public XamlLanguageServerProtocol(
+            [ImportMany] IEnumerable<Lazy<IRequestHandler, IRequestHandlerMetadata>> requestHandlers, 
+            XamlProjectService projectService,
+            [Import(AllowDefault = true)] IXamlLanguageServerFeedbackService? feedbackService)
```

Comment:
❔ Why optional? Are we expecting `XamlLanguageServerProtocol` to be used in scenarios where `IXamlLanguageServerFeedbackService` is not provided?

Reply:
The implementation in xaml language service is not available yet. I will need to check in the interface in Roslyn first and then be able to write the implementation in XLS. Make it optional for now so this current change will not break existing LSP functions.

---
In reply to: [563983456](https://github.com/dotnet/roslyn/pull/50754#discussion_r563983456) [](ancestors = 563983456)